build.gradle.kts

```diff
@@ -9,6 +14,75 @@ allprojects {
     group = findProperty("GROUP").toString()
 }
 
+subprojects {
+    tasks.withType<JavaCompile>().configureEach {
+        sourceCompatibility = JavaVersion.VERSION_17.toString()
+        targetCompatibility = JavaVersion.VERSION_17.toString()
+    }
+
+    tasks.withType<KotlinCompile>().configureEach {
+        kotlinOptions {
+            jvmTarget = JavaVersion.VERSION_17.toString()
+        }
+    }
+
+    plugins.withType<BasePlugin>().configureEach {
+        extensions.configure<BaseExtension> {
+            compileSdkVersion(libs.versions.android.compile.sdk.get().toInt())
+            buildToolsVersion(libs.versions.android.build.tools.get())
+            ndkVersion = libs.versions.android.ndk.get()
+
+            defaultConfig {
+                minSdk = libs.versions.android.min.sdk.get().toInt()
+                targetSdk = libs.versions.android.target.sdk.get().toInt()
+            }
+
+            compileOptions {
+                sourceCompatibility = JavaVersion.VERSION_17
+                targetCompatibility = JavaVersion.VERSION_17
+            }
+        }
+
+        extensions.findByType<LibraryExtension>()?.apply {
+            lint {
+                targetSdk = libs.versions.android.target.sdk.get().toInt()
+            }
+        }
+    }
+
+    plugins.withType<MavenPublishPlugin>().configureEach {
+        extensions.configure<PublishingExtension> {
+            repositories {
+                val repositoryUrl = if (isReleaseBuild) {
+                    requireNotNull(property("RELEASE_REPOSITORY_URL")?.toString())
+                } else {
+                    requireNotNull(property("SNAPSHOT_REPOSITORY_URL")?.toString())
+                }
+
+                maven {
+                    url = uri(repositoryUrl)
+
+                    credentials {
+                        username = findStringPropertyOrDefault("NEXUS_USERNAME")
+                        password = findStringPropertyOrDefault("NEXUS_PASSWORD")
+                    }
+                }
+            }
+        }
+    }
+
+    plugins.withType<SigningPlugin>().configureEach {
```

Comment:
This applies the signing logic consistently across all subprojects which use the `signing` Gradle plugin.